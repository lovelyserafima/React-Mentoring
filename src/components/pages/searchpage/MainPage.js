import React, { Component } from 'react';
import PageName from "../../header/pagename/PageName";
import FormTitle from "../../header/formtitle/FormTitle";
import {HeaderCSSGrid, SearchCSSGrid} from "./MainPage.Styles";
import SearchForm from "../../header/searchform/SearchForm";
import ResultsOptions from "../../helper/resultsoption/search/ResultOptions";
import Results from "../../body/results/Results";
import {sortingTypeForDisplay, sortingTypeForSearch} from "../../constants/CommonConstants";
import { connect } from 'react-redux';
import { getSearchData } from '../../../redux/actions';
import LoadingWrapper from '../../Helper/Loading/Loading';

const mapStateToProps = state => ({
    movies: state.movieReducer.movies,
    sortingType: state.searchReducer.sortingType,
    searchOption: state.searchReducer.searchOption,
    isSearching: state.searchReducer.isSearching,
    error: state.searchReducer.isSearching
});

export class MainPage extends Component {

    performSearch = searchString => {
        const { sortingType, searchOption, getSearchData, error } = this.props;
        console.log(sortingType, searchOption, getSearchData, error);
        getSearchData(
            searchString,
            sortingTypeForSearch[sortingType],
            searchOption
        );
        if (error) {
            console.log('Search failed');
        }
    };

    render() {
        const { movies, sortingType, isSearching } = this.props;
        return (
            <SearchCSSGrid>
                <HeaderCSSGrid>
                    <PageName name={'netflixroulette'} />
                    <FormTitle title={'FIND YOUR MOVIE'} />
                    <SearchForm handleFormSubmit={this.performSearch} />
                </HeaderCSSGrid>
                <ResultsOptions sortingType={sortingTypeForDisplay[sortingType]} />
                {isSearching ? (
                    <LoadingWrapper />
                ) : (
                    movies.length !== 0 && <Results results={movies} />
                )}
            </SearchCSSGrid>
        );
    }
}

export default connect(
    mapStateToProps,
    { getSearchData }
)(MainPage);