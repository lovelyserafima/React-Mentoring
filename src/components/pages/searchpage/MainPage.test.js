import React from 'react';
import { shallow, mount } from 'enzyme';

import MainPage from './MainPage';

describe('SearchPage', () => {
    it('should render correctly', () => {
        const component = shallow(<MainPage />);

        expect(component).toMatchSnapshot();
    });

    it('should render loading spinner when no data is in state', () => {
        const component = shallow(<MainPage />);
        expect(component.find('Results')).toHaveLength(0);
    });

    it('should render results when data for movies is in state', () => {
        const component = shallow(<MainPage />);
        expect(component.find('Results')).toHaveLength(0);
        component.setState({ data: 'movies data' });
        expect(component.find('Results')).toHaveLength(1);
    });

    it('changeSorting changes state when called', () => {
        const component = shallow(<MainPage />);

        component.find('ResultsOptions').prop('changeSorting')('sortingOne');
        expect(component.state('sortingType')).toEqual('sortingOne');
    });

    it('changeSearch changes state when called', () => {
        const component = shallow(<MainPage />);

        component.find('SearchForm').prop('changeSearch')('searchingOne');
        expect(component.state('searchOption')).toEqual('searchingOne');
    });
});