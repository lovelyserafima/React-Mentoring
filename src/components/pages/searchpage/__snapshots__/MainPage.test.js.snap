// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should render correctly 1`] = `
<MainPageStyles__SearchCSSGrid>
  <MainPageStyles__HeaderCSSGrid>
    <PageName
      name="netflixroulette"
    />
    <FormTitle
      title="FIND YOUR MOVIE"
    />
    <SearchForm
      changeSearch={[Function]}
      handleFormSubmit={[Function]}
      searchOption="title"
    />
  </MainPageStyles__HeaderCSSGrid>
  <ResultsOptions
    changeSorting={[Function]}
    dataSize={0}
    sortingType="release date"
  />
  <p>
    loading
  </p>
</MainPageStyles__SearchCSSGrid>
`;
